sdk/python/kfp/_client.py

```diff
@@ -255,6 +257,43 @@ def run_pipeline(self, experiment_id, job_name, pipeline_package_path=None, para
       IPython.display.display(IPython.display.HTML(html))
     return response.run
 
+  def run_pipeline_func_on_kfp(self, pipeline_func: Callable, arguments: Mapping[str, str], run_name=None, experiment_name=None):
```

Comment:
Maybe we can make the function to be more generic so that it can work with func, zip file or pipeline id and we can deprecate the other run method in the same class here. WDYT?

Reply:
I think this would go somewhat against the API parity that @IronPan really wants. He wants the APIs to be exactly the same as the API server APIs (plus maybe some additional high-level "sugar" APIs like this function).

Another problem is that Python does not have good function overloading, so I'd prefer having a separate function for that (maybe with common code).

`run_pipeline_func(func, arguments)`
`run_pipeline_package(pipeline_file, arguments)`
`run_pipeline(pipeline_id, arguments)`

or to be closer to sever API:

`create_run_from_pipeline_func(func, arguments)`
`create_run_from_pipeline_package(pipeline_file, arguments)`
`create_run_from_pipeline_id(pipeline_id, arguments)`

I think that `run_pipeline_func` in this PR covers the 90% case, but I can add other functions in another PR if we get agreement from @IronPan.

WDYT?

Reply:
Talked offline. We've agreed to add the `create_run_from_pipeline_package` function in the next PR.